### PR TITLE
Fix FreeRTOS-Plus device defender, shadow and job demo warnings

### PR DESCRIPTION
**Pull request by chinglee-iot**
Fix multiple demos compilation warning

Description
-----------
* Fix `xPlatformIsNetworkUp()` is not declared compilation warning
* Fix `uxRand()` is not declared compilation warning
* Fix type casting compilation warning

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] ~~I have modified and/or added unit-tests to cover the code changes in this Pull Request.~~

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c [hunk 1]
```diff
@@ -388,7 +388,7 @@ static void prvSendUpdateForJob( char * pcJobId,
 static void prvProcessJobDocument( char * pcJobId,
                                    uint16_t usJobIdLength,
                                    char * pcJobDocument,
-                                   uint16_t jobDocumentLength );
+                                   size_t jobDocumentLength );
```
Comment (ActoryOu) [inline FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c:391]:
```suggestion
                                   size_t uxJobDocumentLength );
```

Should we align the coding style for size_t variables?

**Review by ActoryOu — DISMISSED** [review]

File: FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c [hunk 2]
```diff
@@ -968,9 +971,9 @@ void prvJobsDemoTask( void * pvParameters )
             {
                 /* Handler function to process Jobs message payload. */
                 prvNextJobHandler( pxJobMessagePublishInfo );
-                vPortFree( pxJobMessagePublishInfo->pTopicName );
-                vPortFree( pxJobMessagePublishInfo->pPayload );
-                vPortFree( pxJobMessagePublishInfo );
+                vPortFree( ( char * ) ( pxJobMessagePublishInfo->pTopicName ) );
```
Comment (aggarg) [inline FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c:974]: vPortFree takes a `void *` - https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/portable/MemMang/heap_4.c#L348. Why do we need this cast?

Comment (chinglee-iot) [reply FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c:974]: `pTopicName` and `pPayload` is declared with const prefix. 

```c
{
    ...
    /**
     * @brief Topic name on which the message is published.
     */
    const char * pTopicName;

    /**
     * @brief Length of topic name.
     */
    uint16_t topicNameLength;

    /**
     * @brief Message payload.
     */
    const void * pPayload;

    /**
     * @brief Message payload length.
     */
    size_t payloadLength;
} MQTTPublishInfo_t;
```
`pxJobMessagePublishInfo` is not. Update in new commit.

Comment (aggarg) [reply FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c:974]: Shouldn't that cast be to `void *` in that case, to match the API signature?

**Review by aggarg — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**